src/app/InteractionModelEngine.cpp

```diff
@@ -328,10 +309,10 @@ CHIP_ERROR InteractionModelEngine::OnReadInitialRequest(Messaging::ExchangeConte
             bool keepSubscriptions = true;
             System::PacketBufferTLVReader reader;
             reader.Init(aPayload.Retain());
-            SuccessOrExit(err = reader.Next());
+            ReturnErrorOnFailure(reader.Next());
```

Comment:
Also struggle on this point. I think if server return failure via StatusReponse, the client would get to know what is going on in server side with best effort, and instead of waiting for some time and generate timeout error.

Reply:
Right, the spec says that you're supposed to send a status response if you get a malformed message....

Reply:
ack, thanks